src/transformers/trainer.py

```diff
@@ -3335,7 +3336,7 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
                 torch.mlu.empty_cache()
             elif is_torch_npu_available():
                 torch.npu.empty_cache()
-            elif is_torch_mps_available(min_version="2.0"):
+            elif get_torch_version() >= "2.0" and is_torch_mps_available():
```

Comment:
Good effort on this solution, but it's not quite what we need as that's a string comparator which won't do what we want :) 

The check we want to add to `is_torch_mps_available` is modifying it a bit like so:

```python
def is_torch_mps_available(minimum_version=None):
    if minimum_version is not None:
        ... # do what already exists
    return version.parse(_torch_version) >= version.parse(minimum_version)
```

And now the code we have here should work. 

Does this make sense to follow? :)

Reply:
Thanks for the explanation @muellerzr. I modified the `is_torch_mps_available` function. Please let me know if there is anything else.